Add Release and Source build configurations for ROS 2 Rolling Ridley.

nuclearsandwich opened the pull request:
This adds the release and source build configurations in order to
prepare the distribution for bootstrap. I'll port CI job configurations
from Foxy as a follow-up next week.

Signed-off-by: Steven! Ragnarök <steven@nuclearsandwich.com>

nuclearsandwich committed:
Add Release and Source build configurations for ROS 2 Rolling Ridley.

This adds the release and source build configurations in order to
prepare the distribution for bootstrap. I'll port CI job configurations
from Foxy as a follow-up next week.

Signed-off-by: Steven! Ragnarök <steven@nuclearsandwich.com>

dirk-thomas commented on index.yaml:
```diff
@@ -73,6 +73,17 @@ distributions:
       ubv8: foxy/release-focal-arm64-build.yaml
     source_builds:
       default: foxy/source-build.yaml
+  foxy:
```
Copy-n-paste mistake.

nuclearsandwich replied:
Had to be one. :wink:

dirk-thomas commented on index.yaml:
```diff
@@ -73,6 +73,17 @@ distributions:
     source_builds:
       default: foxy/source-build.yaml
+  foxy:
+    ci_builds: {}
```
I assume you plan to add them after the first round of Debian packages has been built successfully>

nuclearsandwich replied:
Your assumption is correct. The pull request description includes "I'll port CI job configurations from Foxy as a follow-up next week." as an indication toward this.

dirk-thomas commented on rolling/release-build.yaml:
```diff
@@ -0,0 +1,94 @@
+build_environment_variables:
+  RTI_NC_LICENSE_ACCEPTED: 'yes'
+
+jenkins_binary_job_priority: 78
```
The priorities should be different from Foxy.

Same below.

tfoote replied:
We should set the rolling distro to lower priority than all the other distros. If it's a little slower rebuilding that's ok. We don't want to worry about gating rolling releases if we're in a push for a stable fixup.

dirk-thomas replied:
I would say the opposite of also true: we don't want a blocked rolling rebuild lead to failures on PRs.

nuclearsandwich replied:
My first instinct was also to de-prioritize the rolling distribution in favor of stable ones. Which is what I did in e3d9fe6.

dirk-thomas replied:
Maybe either of you can clarify your rational. I would think we want changes in the rolling distro to turn around asap since that is where the majority of development will happen so a quick response time on PR is important.

nuclearsandwich replied:
> Maybe either of you can clarify your rational.

I think we might have different primary audiences in mind. I'm thinking mostly of the fact that community-wide, the stable distributions are more important and are at a higher support tier, therefore have priority.

I almost made the rolling PR jobs higher priority than stable PR jobs but it would require juggling a lot of the existing priorities and I am more focused on getting the distro bootstrapped than on settling the running priorities right now.

dirk-thomas replied:
> I'm thinking mostly of the fact that community-wide, the stable distributions are more important and are at a higher support tier, therefore have priority.

I don't see how the user perspective priority is relevant here. Binary packages for released distros happen whenever the maintainer does them and should sit in the testing repo for some time for adventures users to try them before they are being synced to the main repo. So I would argue if it takes a day longer to get into testing that isn't a big deal since we shouldn't sync right to main anyway after a rebuild.

> I am more focused on getting the distro bootstrapped than on settling the running priorities right now.

Please feel free to merge this as-is to get started with `rolling` but make sure that we are following up on this topic and persist the rational and conclusion somewhere (and if necessary update priorities is they change).

nuclearsandwich replied:
> So I would argue if it takes a day longer to get into testing that isn't a big deal since we shouldn't sync right to main anyway after a rebuild.

This is most often the case however it's not uncommon for releases which resolve known regressions to go in just before a sync during the freeze window.

> make sure that we are following up on this topic and persist the rational and conclusion somewhere 

That's reasonable to me. @tfoote do you want to elaborate on your thought process and I'll start incorporating this into documentation of the status quo which we can use as a basis for PRs with motivated changes.

nuclearsandwich committed:
foxy -> rolling.

I do not think anything could be more Friday afternoon than this commit.

nuclearsandwich committed:
Update priority levels for rolling jobs.

I opted to make the rolling jobs lower priority than the stable
distributions. The rolling jenkins_pull_request_job_priority for rolling
is at the same priority level as foxy CI jobs but they run on different
agents so that's not a problem.
